lib/tasks/db.rake

```diff
@@ -295,6 +295,292 @@ task 'db:stats' => 'environment' do
   print_table(DB.query_hash(sql))
 end
 
+class TemporaryDB
+  PG_PATH = "/tmp/pg_schema_tmp"
+  PG_CONF = "#{PG_PATH}/postgresql.conf"
+  PG_SOCK_PATH = "#{PG_PATH}/sockets"
+
+  def port_available?(port)
+    TCPServer.open(port).close
+    true
+  rescue Errno::EADDRINUSE
+    false
+  end
+
+  def pg_bin_path
+    return @pg_bin_path if @pg_bin_path
+
+    ["13", "12", "11", "10"].each do |v|
+      bin_path = "/usr/lib/postgresql/#{v}/bin"
+      if File.exist?("#{bin_path}/pg_ctl")
+        @pg_bin_path = bin_path
+        break
+      end
+    end
+    if !@pg_bin_path
+      bin_path = "/Applications/Postgres.app/Contents/Versions/latest/bin"
+      if File.exists?("#{bin_path}/pg_ctl")
+        @pg_bin_path = bin_path
+      end
+    end
+    if !@pg_bin_path
+      puts "Can not find postgres bin path"
+      exit 1
+    end
+    @pg_bin_path
+  end
+
+  def initdb_path
+    return @initdb_path if @initdb_path
+
+    @initdb_path = `which initdb 2> /dev/null`.strip
+    if @initdb_path.length == 0
+      @initdb_path = "#{pg_bin_path}/initdb"
+    end
+
+    @initdb_path
+  end
+
+  def find_free_port(range)
+    range.each do |port|
+      return port if port_available?(port)
+    end
+  end
+
+  def pg_port
+    @pg_port ||= find_free_port(11000..11900)
+  end
+
+  def pg_ctl_path
+    return @pg_ctl_path if @pg_ctl_path
+
+    @pg_ctl_path = `which pg_ctl 2> /dev/null`.strip
+    if @pg_ctl_path.length == 0
+      @pg_ctl_path = "#{pg_bin_path}/pg_ctl"
+    end
+
+    @pg_ctl_path
+  end
+
+  def start
+    FileUtils.rm_rf PG_PATH
```

Comment:
Minor would name this something like `PG_TMP_PATH`. On the first look I thought we were moving a folder within the PG data directory.

Reply:
sure renaming

Reply:
sure done.

Reply:
actually it kind of complicates the code a tiny bit, going to keep ti using pg ctl for now. 

worse case it breaks one day and just hangs there. will not be a giant problem.